### ISSUE TITLE
typing-extensions dependency

### ISSUE THREAD

**Issue by WernerHu** [DisasterAWARE/aws-glue-schema-registry-python#8]
Hi @cdebost, 

first of all thanks for your great work. 
We would like to use your library within a AWS managed Airflow (MWWA). Because of that we're bound to Python 3.7 and have a [restriction](https://raw.githubusercontent.com/apache/airflow/constraints-2.0.2/constraints-no-providers-3.7.txt) of the typing-extensions to 3.7.4.3 

Is there a specific reason why this library has a dependency to typing-extensions>=3.10.0.2;python_version<"3.8"?
If not, I can adjust the dependency and create a pr.

Thanks 

BR

**Comment by cdebost** [comment]
Hi @WernerHu.

The only reason I added a dependency on `typing-extensions` is for `typing.Protocol` which isn't in python 3.7. As for why I did `>=3.10.02`, I think it's just because that was the latest available version of `typing-extensions` at the time.

If I understand you correctly, you want to lower that minimum version to `>=3.7.4.3` - if you are willing to test that downgrading that version doesn't cause any issues (I doubt it would), I have no problem merging that change.
